plugins/nf-quilt/src/main/nextflow/quilt/QuiltProduct.groovy

```diff
@@ -117,43 +117,29 @@ ${nextflow}
     private final Session session
     private String msg
     private Map meta
-    final Integer pubStatus
 
     QuiltProduct(QuiltPath path, Session session) {
         this.path = path
         this.pkg = path.pkg()
         this.msg =  pkg.toString()
         this.meta = [pkg: msg, time_start: now()]
         this.session = session
-        this.pubStatus = -1
         if (session.isSuccess() || pkg.is_force()) {
-            try {
-                this.pubStatus = publish()
-            }
-            catch (Exception e) {
-                log.error("publish failed: ${e.getMessage()}", pkg.meta)
-            }
+            publish()
         } else {
             log.info("not publishing: ${pkg} [unsuccessful session]")
```

Comment:
I think this should throw an exception, too, but I don't know which one

Reply:
Some options

[Sync­Failed­Exception](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/SyncFailedException.html) ✔

[Protocol­Exception](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/ProtocolException.html) ✔

[Server­Exception](https://docs.oracle.com/en/java/javase/11/docs/api/java.rmi/java/rmi/ServerException.html) ✔

Maybe define your own?

Reply:
All of those are for very specific Java features, so doesn't seem right. They all inherit from `IOException`, so could just use that one.

But, `IOException` is for failures that the user has no control over (network failures, etc.) - while here, it looks like the user should've checked that the session is valid. If that's the case, then `IllegalStateException` might be more appropriate.